File: Go/Algorithms/shuttleSort.go
package main

func ShuttleSort(array []int) []int {
	for i := 1; i < len(array); i++ {
		j := i
		for j > 0 && array[j-1] > array[j] {
			temp := array[j]
			array[j] = array[j-1]
			array[j-1] = temp
			j--
		}
	}
	return array
}

File: Lua/Algorithms/shuttleSort.lua
function shuttleSort(array)
    for i = 2, #array, 1 do
        local j = i

        while j > 1 and array[j-1] > array[j] do
            local temp = array[j]
            array[j] = array[j-1]
            array[j-1] = temp
             j = j - 1
        end
    end
    return array
end

array = {5, 4, 3, 2, 1}

print(table.concat(shuttleSort(array), ", "))